corgidrp/walker.py

```diff
@@ -235,16 +237,49 @@ def guess_template(dataset):
         elif image.pri_hdr['VISTYPE'] == "FFIELD":
             recipe_filename = "l1_flat_and_bp.json"
         elif image.pri_hdr['VISTYPE'] == "DARK":
-            recipe_filename = "l1_to_l2a_noisemap.json"
+            _, unique_vals = dataset.split_dataset(exthdr_keywords=['EXPTIME', 'CMDGAIN', 'KGAIN'])
+            if 'ISPC' in image.ext_hdr: # to be added eventually, so check if it's present first
+                if image.ext_hdr['ISPC']:
+                    recipe_filename = "l1_to_l2b_pc_dark.json"
+                elif len(unique_vals) > 1: # darks for noisemap creation
+                    recipe_filename = "l1_to_l2a_noisemap.json"
+                elif len(unique_vals) == 1: # traditional darks
+                    recipe_filename = "build_trad_dark_image.json"
+            else: # can't distinguish PC or analog then; default to analog
+                if len(unique_vals) > 1: # darks for noisemap creation
+                    recipe_filename = "l1_to_l2a_noisemap.json"
+                elif len(unique_vals) == 1: # traditional darks
+                    recipe_filename = "build_trad_dark_image.json"
         elif image.pri_hdr['VISTYPE'] == "PUPILIMG":
             recipe_filename = ["l1_to_l2a_nonlin.json", "l1_to_kgain.json"]
         else:
-            recipe_filename = "l1_to_l2b.json"
+            if 'ISPC' in image.ext_hdr:
+                if image.ext_hdr['ISPC']:
```

Comment:
can this be done as `if 'ISPC' in image.ext_hdr & image.ext_hdr['ISPC']`? The second have of the conditional will only be run if the first half is True. 

Right now, if "ISPC" is in the header and it is False, the walker does nothing, which is not what we want (I assume we just want it to run "l1_to_2b.json". Same later on below.